src/Temporalio/Worker/WorkflowInstance.cs

```diff
@@ -1402,6 +1385,71 @@ private string GetStackTrace()
             }).Where(s => !string.IsNullOrEmpty(s)).Select(s => $"Task waiting at:\n{s}"));
         }
 
+        private void ApplyCompletionCommandReordering(
+            WorkflowActivation act, WorkflowActivationCompletion completion)
+        {
+            // In earlier versions of the SDK we allowed commands to be sent after workflow
+            // completion. These ended up being removed effectively making the result of the
+            // workflow function mean any other later coroutine commands be ignored. To match
+            // Go/Java, we are now going to move workflow completion to the end (well, before any
+            // query results) so that same-task-post-completion commands are processed.
+            //
+            // Note this only applies for successful activations that don't have completion
+            // reordering disabled and that are either not replaying or have the flag set.
+            if (completion.Successful == null || disableCompletionCommandReordering)
+            {
+                return;
+            }
+            if (IsReplaying && !act.AvailableInternalFlags.Contains((uint)WorkflowLogicFlag.ReorderWorkflowCompletion))
+            {
+                return;
+            }
+
+            // We know we're on a newer SDK and can move completion to the end if we need to. First,
+            // find the completion command and the number of trailing query commands.
+            var completionCommandIndex = -1;
+            var trailingQueryCommandCount = 0;
```

Comment:
Yes, I know technically I could have just moved past trailing query commands (and not queries come in their own activation IIRC), but in my head it only makes sense to move until before those. But I can simplify the logic by just always moving to the end if it others think it's worth it.

Reply:
Yeah I just wouldn't bother with this

Reply:
Concur, will simplify

Reply:
Fixed, made logic a tad simpler